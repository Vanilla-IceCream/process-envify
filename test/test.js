import envify from '../src';

process.chdir(__dirname);

describe('process-envify', () => {
  it('should inject strings (process.env)', () => {
    const data = {
      BOOK_NAME: 'ECMAScript: Up and Running',
    };

    expect(envify(data)).toMatchSnapshot();
  });

  it('should inject strings (import.meta)', () => {
    const data = {
      BOOK_NAME: 'ECMAScript: Up and Running',
    };

    expect(envify(data, { useImportMeta: true })).toMatchSnapshot();
  });
});
