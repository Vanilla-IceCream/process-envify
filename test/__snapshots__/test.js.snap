// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process-envify should inject strings (import.meta) 1`] = `
Object {
  "import.meta.env.BOOK_NAME": "\\"ECMAScript: Up and Running\\"",
}
`;

exports[`process-envify should inject strings (process.env) 1`] = `
Object {
  "process.env.BOOK_NAME": "\\"ECMAScript: Up and Running\\"",
}
`;
